server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1293,6 +1295,33 @@ public static Setting<Float> floatSetting(
         );
     }
 
+    private class FloatParser implements Function<String, Float>, Writeable {
```

Comment:
You need to actually create an instance of this parser when creating a `floatSetting()` (line 1357) similar to how you've done for the other number parsers.

(Style) For consistency try to put the parsers in the same locations relative to where they are used.  The Integer, Long, and Double parsers are all after they are used, so this one should probably match to be consistent.